test/release/examples/primers/modules.chpl

```diff
@@ -0,0 +1,386 @@
+         naming the full path to reach it.  Otherwise, there is no difference
+         between the use statement for a top-level module and that of a nested
+         module.
+
+         (see the module ContainsNesting and its submodules at the end of this
+         file for an example of a nested module)
+      */
+
+      use ContainsNesting.Inner1;
+
+      writeln(foobar); // Will output Inner1.foobar, or '14'
+    }
+
+    {
+      /* Multiple modules may be used in the same use statement */
+      use modToUse, AnotherModule, ThirdModule;
+
+      if (a || b < 0) {
+        // Refers to AnotherModule.a (which is 'false') and ThirdModule.b (which
+        // is '-13.0')
+        writeln(foo); // Refers to modToUse.foo
+      } else {
+        writeln(bar); // Refers to modToUse.bar
+      } // Will output modToUse.foo (which is '12')
+    }
+
+
+    {
+      /* Equivalently, a scope may contain multiple use statements */
+      use modToUse;
+      use AnotherModule, ThirdModule;
+
+      writeln(a && foo > 15);
+      // outputs false (because AnotherModule.a is 'false' and modToUse.foo is
+      // '12')
+    }
+
+
+    {
+      /* In either case, the modules used in this way are considered in concert
+         (after symbols defined at this scope but before symbols defined outside
+         of it) - the ordering within a use statement or across multiple use
+         statements does not affect the precedence of symbols that share a name.
+         This means that if two modules each define a symbol with the same name,
+         and both modules are used at the same scope, attempts to access a
+         symbol by that name will result in a naming conflict:
+      */
+      use modToUse, Conflict;
+
+      writeln(foo); // Outputs modToUse.foo ('12')
+      /* The following line would fail because both modToUse and Conflict define
+         a symbol named bar:
+      */
+      // writeln(bar);
+      writeln(other); // Outputs Conflict.other ('5.0 + 3.0i')
+    }
+
+    {
+      /* When the symbol being accessed is the name of a function, the rules
+         become more complex.  If the two function definitions are overloads (or
+         define different arguments), then the best match will be found, no
+         matter where the function is defined relative to the other function
+         definitions.
+
+         More details on when overloading applies, when functions may shadow
+         other functions, etc. can be found in the relevant section of the
+         language specification.  They will not be covered further in this
+         primer.
+      */
+
+      use modToUse, DifferentArguments;
+
+      writeln(baz(2, 3));
+      // Accesses the function modToUse.baz using the two arguments.  Should
+      // output 2 * (2 + 3) or '10'
+      writeln(baz(3));
+      // Access the function DifferentArguments.baz using the single argument.
+      // Should output 3 - 2, or '1'
+    }
```

Comment:
I prefer defining it here (where I've made it clear that multiple modules may be used) so that I don't have to define a function in scope to show how multiple functions with the same name interact.

Additionally, I'd prefer not to introduce a more complicated topic earlier in the primer when the reader doesn't know what rules might be affected by it, as they are more likely to just forget about the additional case because they've been told not to worry about it.  All the new rules after this point (except enums) apply to both types of symbol resolution so it's okay if they forget about how functions differ when reading about the new rules.